go/cmd/vtbackup/vtbackup.go

```diff
@@ -362,7 +362,7 @@ func takeBackup(ctx context.Context, topoServer *topo.Server, backupStorage back
 		case <-time.After(time.Second):
 		}
 
-		status, statusErr := mysqld.ReplicationStatus()
+		status, statusErr := mysqld.ReplicationStatus(ctx)
```

Comment:
cases like these are what scare me a bit in that we didn't have a timeout at all before and are now passing in whatever context was in scope. i think the only alternative would be to use `Context.TODO` in callsites that are unrelated to the our replication polling change but that sounds like a pain to keep track of, as i think it's directionally correct to have a context with a reasonable timeout throughout.

i think that's all to say: we should keep an eye on this other uses of the replication status check to ensure they continue to work as expected as we roll this out